src/resource-select/ResourceSelect.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { Select } from 'antd';
+import { observer } from 'mobx-react';
+import { ResourceCollection } from 'webpanel-data';
+import { FormElementProps } from '../form/form/FormElement';
+// import { ResourceCollectionConfig } from 'webpanel-data/lib/ResourceCollection';
+
+import './ResourceSelect.css';
+
+export interface ResourceSelectProps extends FormElementProps {
+  resourceCollection: ResourceCollection;
+  displayedField: string;
+}
+
+export interface ResourceSelectState {
+  options: object[];
+  value: string;
+}
+
+@observer
+export class ResourceSelect extends React.Component<ResourceSelectProps, ResourceSelectState > {
+  state = {
+    options: [],
+    value: '',
+  };
+
+  componentDidMount() {
+
+    const resource = this.props.resourceCollection;
+    resource.get().catch(this.handleError);
+
+    const interval = setInterval(() => {
+      if (!resource || !resource.loading) {
+        this.setState({
+          options: [...resource.data || []],
+        });
+        clearInterval(interval);
+      }
+    }, 100);
+  };
+
+  handleError(err: Error) {
+    console.error(err);
+  }
+
+  onSearch = (value: string) => this.setState({value});
+
+  onChange = (value: string) => this.setState({value});
+
+  render() {
+    const displayedField: string = this.props.displayedField;
+
+    const options = this.state.options.map((proxy: any) => {
+        if (
+          proxy[displayedField] !== undefined &&
+          proxy[displayedField]
+            .toString()
+            .toLowerCase()
+            .indexOf(this.state.value.toLowerCase()) > -1
+        ) {
+          return <Select.Option
+            key={proxy[displayedField]}
+            value={proxy[displayedField]}
+          >
+            {proxy[displayedField]}
+          </Select.Option>
+        } else {
+          return null;
+        }
+      }
+    );
```

Comment:
this actually searches through all values from resource collection. But the scenario should be:
1) on load check if the select has any initial value and if:
a) it hasn't any value, display empty option list
b) it has any value, filter the collection by {id:value} and display the options
2) on search remove the filter by id (unset filters) and use `updateSearch` to use filter by entered string